File: src/dotnet/Common/Models/Orchestration/OpenAITextMessageContentItem.cs
using System.Text.Json.Serialization;
using FoundationaLLM.Common.Constants.Orchestration;

namespace FoundationaLLM.Common.Models.Orchestration
{
    /// <summary>
    /// An OpenAI text message content item.
    /// </summary>
    public class OpenAITextMessageContentItem : MessageContentItemBase
    {
        /// <inheritdoc/>
        [JsonIgnore]
        public override string? Type { get; set; }

        /// <summary>
        /// A list of file annotations used to generate the message content item.
        /// </summary>
        [JsonPropertyName("annotations")]
        public List<OpenAIFilePathContentItem> Annotations { get; set; } = [];

        /// <summary>
        /// The text value of the message content item.
        /// </summary>
        [JsonPropertyName("value")]
        public string? Value { get; set; }

        /// <summary>
        /// Set default property values.
        /// </summary>
        public OpenAITextMessageContentItem() =>
            Type = MessageContentItemTypes.Text;
    }
}

File: src/python/PythonSDK/foundationallm/models/orchestration/openai_text_message_content_item.py
from pydantic import Field
from typing import Optional, List
from .message_content_item_base import MessageContentItemBase
from .openai_file_path_message_content_item import OpenAIFilePathMessageContentItem
from .message_content_item_types import MessageContentItemTypes

class OpenAITextMessageContentItem(MessageContentItemBase):
    """An OpenAI text message content item."""
    
    annotations: Optional[List[OpenAIFilePathMessageContentItem]] = Field(default_factory=list, alias="annotations")
    value: Optional[str] = Field(None, alias="value")
    
    def __init__(self, **data):
        super().__init__(**data)
        self.type = MessageContentItemTypes.TEXT
